clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureNetworkClient.groovy

```diff
@@ -30,38 +37,213 @@ class AzureNetworkClient extends AzureBaseClient {
+   * @param loadBalancerName name of the load balancer to delete
+   * @return an OperationResponse object
+   */
+  OperationResponse deleteLoadBalancer(AzureCredentials creds, String resourceGroupName, String loadBalancerName) {
+    // First delete the public Ip associated with the load balancer
+    def loadBalancer = getNetworkResourceProviderClient(creds).getLoadBalancersOperations().get(resourceGroupName, loadBalancerName).getLoadBalancer()
+
+    if (loadBalancer.frontendIpConfigurations.size() != 1) {
+      throw new RuntimeException("Unexpected number of public IP addresses associated with the load balancer (should be only one)!")
+    }
+
+    def publicIpAddressName = AzureUtilities.getResourceNameFromID(loadBalancer.frontendIpConfigurations.first().getPublicIpAddress().id)
+    this.getNetworkResourceProviderClient(creds).getPublicIpAddressesOperations().delete(resourceGroupName, publicIpAddressName)
+
+    this.getNetworkResourceProviderClient(creds).getLoadBalancersOperations().delete(resourceGroupName, loadBalancerName)
+  }
+
+  /**
+   *
+   * @param creds the credentials to use when communicating to the Azure subscription(s)
+   * @param resourceGroupName name of the resource group where the load balancer was created
+   * @param virtualNetworkName name of the virtual network to create
+   * @param region region to create the resource in
+   */
+  void createVirtualNetwork(AzureCredentials creds, String resourceGroupName, String virtualNetworkName, String region, String addressPrefix = "10.0.0.0/16") {
+    try {
+
+      def virtualNetwork = new VirtualNetwork(region)
+      AddressSpace addressSpace = new AddressSpace()
+      addressSpace.addressPrefixes.add(addressPrefix)
+      virtualNetwork.setAddressSpace(addressSpace)
+
+      //Create the virtual network for the resource group
+      AzureAsyncOperationResponse response = this.getNetworkResourceProviderClient(creds).
+        getVirtualNetworksOperations().
+        createOrUpdate(resourceGroupName, virtualNetworkName, virtualNetwork)
+    }
+    catch (e) {
+      throw new RuntimeException("Unable to create Virtual network ${virtualNetworkName} in Resource Group ${resourceGroupName}", e)
+    }
+  }
+  /**
+   * Retrieve a collection of all subnets for a give set of credentials, regardless of resource group/region
+   * @param creds the credentials to use when communicating to the Azure subscription(s)
+   * @return a Collection of objects which represent a Subnet in Azure
+   */
+  Collection<AzureSubnetDescription> getSubnetsAll(AzureCredentials creds) {
+    def list = this.getNetworkResourceProviderClient(creds).getVirtualNetworksOperations().listAll().virtualNetworks
+
+    def result = new ArrayList<AzureSubnetDescription>()
+
+    for (VirtualNetwork item : list) {
+      for (com.microsoft.azure.management.network.models.Subnet itemSubnet : item.subnets) {
```

Comment:
But is it used anywhere in this class? Disambiguation should be covered by the import statements.